### ISSUE TITLE
macOS build fails "ld: library not found for -lminizip" even though CMake has found it

### ISSUE THREAD

**Issue by MarkCallow** [assimp/assimp#3933]
**Describe the bug**
After CMake configure and generate the build fails with the error `ld: library not found for -lminizip`. As can be seen in the screen shot below CMake has found libminizip.dylib. And as can be seen in the link command given in Additional Context there is no option to add /opt/local/lib as a place to search for libaries. So it is not surprising the link fails.

**To Reproduce**
1. Install minizip via macports - `sudo port install minizip`
2. Run CMake GUI. Configure - selecting default build tools (Xcode), and selecting only importers (no tools, no exporters).
3. Generate project
4. `cd <directory with generated project>; cmake --build . --config Release `
5. See error

**Expected behavior**
I expect the build to succeed.

**Screenshots**
Screenshot of CMake GUI showing it found libminizip.dylib in /opt/local/lib:

<img width="825" alt="Screen Shot 2021-05-30 at 9 59 23 PM" src="https://user-images.githubusercontent.com/2244683/120105034-32253680-c10c-11eb-8582-496705e0f389.png">

**Platform (please complete the following information):**
 - OS: macOS
 - Browser: n/a
 - Version 11.4 (Big Sur)

**Additional context**
Here is the link command and output:

```
Ld /Users/mark/Projects/github/assimp/macOS/bin/Release/libassimp.5.0.1.dylib normal (in target 'assimp' from project 'Assimp')
    cd /Users/mark/Projects/github/assimp/assimp
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang++ -target x86_64-apple-macos11.3 -dynamiclib -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk -L/Users/mark/Projects/github/assimp/macOS/bin/Release -F/Users/mark/Projects/github/assimp/macOS/bin/Release -filelist /Users/mark/Projects/github/assimp/macOS/code/Assimp.build/Release/assimp.build/Objects-normal/x86_64/assimp.5.0.1.LinkFileList -install_name /libassimp.5.0.1.dylib -Xlinker -rpath -Xlinker /Users/mark/Projects/github/assimp/macOS/bin/Release -dynamiclib -Wl,-headerpad_max_install_names -install_name @rpath/libassimp.5.dylib /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.3.sdk/usr/lib/libz.tbd /Users/mark/Projects/github/assimp/macOS/bin/Release/libdraco.1.dylib -lminizip -Xlinker -no_adhoc_codesign -compatibility_version 5.0.0 -current_version 5.0.1 -Xlinker -dependency_info -Xlinker /Users/mark/Projects/github/assimp/macOS/code/Assimp.build/Release/assimp.build/Objects-normal/x86_64/assimp.5.0.1_dependency_info.dat -o /Users/mark/Projects/github/assimp/macOS/bin/Release/libassimp.5.0.1.dylib
ld: library not found for -lminizip
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

**Comment by MarkCallow** [comment]
Can I get some help here. I am completely stalled.

I started trying to update my libassimp build because the latest clang versions give lots of warnings about deprecated copy constructors with the assimp headers of the version I was using. Unfortunately, but not surprisingly, if I compile my app with the latest headers but link it to the older version of the library, it crashes when run.

**Comment by MarkCallow** [comment]
Can anyone build this library for macOS? I'm still failing. Don't the authors care?

**Comment by ryandesign** [comment]
Yeah if assimp wants to link with MacPorts minizip, it's going to have to use either `-L/opt/local/lib -lminizip`, or use `-lminizip` and have `/opt/local/lib` in `LIBRARY_PATH`, or use the full path `/opt/local/lib/libminizip.dylib`. I don't know how assimp is deciding what flags to use, but it would have gotten a working set of flags if it had used, say, `pkg-config`:

```
$ pkg-config --libs minizip
-L/opt/local/lib -lminizip
```

If you want something that just works, you can [install assimp with MacPorts](https://ports.macports.org/port/assimp).

**Comment by tellypresence** [comment]
@MarkCallow Is this still a problem?

**Comment by MarkCallow** [comment]
> [@MarkCallow](https://github.com/MarkCallow) Is this still a problem?

I don't use MacPorts any more so I can't say. I'm using vcpkg now for my dependencies, so I get libraries that can be bundled with my app, and vcpkg is building assimp without a problem.

I'll close this as "not planned."
